R/tar_funcs.R

```diff
@@ -0,0 +1,846 @@
+
+  cbind(site_id = pt$site_id, chirps_pt) %>%
+    select(-ID) %>%
+    pivot_longer(
+      -site_id,
+      names_to = "date",
+      values_to = "precip_daily"
+    )
+}
+
+
+
+
+#' load_cccm_wb
+#'
+#' @param path \code{character} path from AA_DATA_DIR to cccm xlsx wb
+#'
+#' @return list of named data.frames each containing 1 sheet from wb
+
+load_cccm_wb <- function(path) {
+  sheet_names <- excel_sheets(path)
+  ss_skip_vals <- c(1, rep(0, 3))
+  wb <- sheet_names %>%
+    map2(
+      .y = ss_skip_vals,
+      ~ read_excel(path, sheet = .x, skip = .y) %>%
+        clean_names()
+    ) %>%
+    set_names(sheet_names)
+}
+
+
+
+#' get_site_locs
+#'
+#' @param wb workbook list object created from `load_cccm_wb()`
+#'
+#' @return sf spatial data frame containing unique sites with coordinates in epsg:4326
+
+get_site_locs <- function(wb) {
+  master <- wb[["ML- Flooding Available data"]]
+  flood_reports <- wb[["CCCM FLOOD REPORT IN IDP SITES"]]
+
+  master_w_coords <- master %>%
+    rename(
+      longitude = "available_coordinates_longitude",
+      latitude = "available_coordinates_latitude"
+    ) %>%
+    filter(!is.na(longitude), !is.na(latitude))
+
+  lat_lon_lookup <- master_w_coords %>%
+    select(
+      site_id,
+      longitude,
+      latitude,
+      site_population
+    )
+
+  sites_ids_w_loc_in_master <- flood_reports %>%
+    rename(site_id = "cccm_idp_sites_master_list_site_id") %>%
+    filter(site_id %in% master_w_coords$site_id) %>%
+    distinct(site_id)
+
+
+  st_as_sf(
+    sites_ids_w_loc_in_master %>%
+      left_join(lat_lon_lookup,
+        by = "site_id"
+      ),
+    coords = c("longitude", "latitude"),
+    crs = 4326
+  )
+}
+
+#' clean_cccm_impact_data
+#'
+#' @param wb workbook list object created from `load_cccm_wb()`
+#' @param floodsites output from `get_site_locs()` (`cccm_flood_report_sites`)
+#' @return flood report data with some minor cleaning and columns added (obj: `cccm_flood_impact_data`)
+
+clean_cccm_impact_data <- function(wb, floodsites) {
+  wb[["CCCM FLOOD REPORT IN IDP SITES"]] %>%
+    rename(
+      site_id = "cccm_idp_sites_master_list_site_id",
+      site_name = "cccm_idp_sites_master_list_site_name"
+    ) %>%
+    filter(site_id %in% floodsites$site_id) %>%
+    left_join(floodsites %>%
+      select(site_id, site_population)) %>%
+    rename(
+      num_init_hhs = "initial_figures_h_hs",
+      num_verified_hhs = "verified_figures_h_hs",
+      num_shelters_affected = "how_many_shelters_were_affected",
+      num_hhs_lost_documentation = "if_yes_how_many_h_hs_did_they_lose_their_personal_legal_documentations_due_to_the_flooding",
+    ) %>%
+    mutate(
+      # https://fscluster.org/sites/default/files/documents/operational_guidance_note_-_minimum_expenditure_basket_september_2022.pdf
+      pop_affected = num_shelters_affected * 7,
+      pct_pop_affected = pop_affected / site_population,
+      ratio_shelters_to_pct_pop = num_shelters_affected / (10 - pct_pop_affected)
```

Comment:
ha the math doesn't make a lot of sense. I just wanted to highlight sites that had both high number of shelters affected and high numbers of % pop affected later in a plot. I had to do `10 - pct_pop_affected` because the `pct_pop_affected` was often much greater than 100 % but less than 1000 % .   By doing `10-pct_pop_affected` on the denominator it just made sure that the ration would keep increasing as the `pct_pop_affected` increased even if it was greater than 100% .

All this was just for early stages when we were coming up with our own ways to prioritize locations just to have something to show and start understanding the data. I don't think it will be used in the grand scheme.